lonestar/analytics/cpu/jaccard/jaccard_cli.cpp

```diff
@@ -99,75 +64,58 @@ main(int argc, char** argv) {
-    GALOIS_LOG_FATAL("could not make property graph: {}", pg_result.error());
-  }
-  Graph graph = pg_result.value();
+  std::cout << "Read " << pfg->topology().num_nodes() << " nodes, "
+            << pfg->topology().num_edges() << " edges\n";
 
-  std::cout << "Read " << graph.num_nodes() << " nodes, " << graph.num_edges()
-            << " edges\n";
-
-  if (base_node >= graph.size() || report_node >= graph.size()) {
+  if (base_node >= pfg->topology().num_nodes() ||
+      report_node >= pfg->topology().num_nodes()) {
     std::cerr << "failed to set report: " << report_node
               << " or failed to set base: " << base_node << "\n";
     abort();
   }
 
-  auto it = graph.begin();
-  std::advance(it, base_node.getValue());
-  GNode base = *it;
-  it = graph.begin();
-  std::advance(it, report_node.getValue());
-  GNode report = *it;
-
   galois::reportPageAlloc("MeminfoPre");
 
   galois::StatTimer execTime("Timer_0");
   execTime.start();
 
-  algo(&graph, base);
+  if (auto r = galois::analytics::Jaccard(
+          pfg.get(), base_node, output_property_name,
+          galois::analytics::JaccardPlan());
+      !r) {
+    GALOIS_LOG_FATAL(
+        "Jaccard failed: {} {}", r.error().category().name(),
+        r.error().message());
+  }
 
   execTime.stop();
 
   galois::reportPageAlloc("MeminfoPost");
 
+  auto pg_result = galois::graphs::PropertyGraph<NodeData, EdgeData>::Make(
+      pfg.get(), {output_property_name}, {});
+  if (!pg_result) {
+    GALOIS_LOG_FATAL("could not make property graph: {}", pg_result.error());
+  }
+  Graph graph = pg_result.value();
+
   std::cout << "Node " << report_node << " has similarity "
-            << graph.GetData<NodeValue>(report) << "\n";
-
-  // Sanity checking code
-  galois::GReduceMax<double> max_similarity;
-  galois::GReduceMin<double> min_similarity;
-  max_similarity.reset();
-  min_similarity.reset();
-
-  galois::do_all(
-      galois::iterate(graph),
-      [&](const GNode& i) {
-        double similarity = graph.GetData<NodeValue>(i);
-        if ((unsigned int)i != (unsigned int)base) {
-          max_similarity.update(similarity);
-          min_similarity.update(similarity);
-        }
-      },
-      galois::loopname("Sanity check"), galois::no_stats());
-
-  galois::gInfo(
-      "Maximum similarity (excluding base) is ", max_similarity.reduce());
-  galois::gInfo("Minimum similarity is ", min_similarity.reduce());
-  galois::gInfo("Base similarity is ", graph.GetData<NodeValue>(base));
-
-  // TODO: Verify?
+            << graph.GetData<NodeValue>(report_node) << "\n";
+
+  auto stats_result = galois::analytics::JaccardStatistics::Compute(
+      pfg.get(), base_node, output_property_name);
+  if (!stats_result) {
+    GALOIS_LOG_FATAL(
+        "could not make compute statistics: {}", stats_result.error());
+  }
+
+  stats_result.value().Print();
 
   if (!skipVerify) {
-    if (graph.GetData<NodeValue>(base) == 1.0) {
+    if (galois::analytics::JaccardAssertValid(
```

Comment:
Isn't this returning `Galois::Result`, so should check `.value`?

Reply:
The assert valid routines don't return a bool any more (at the request of dnn0). They just error out if the checks see invalidities. So this is correct now.

Reply:
Got it. I didn't look at the parallelSTL loop. Makes sense.